aspnetcore/fundamentals/http-requests/samples/GitHub/GitHubService.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace HttpClientFactorySample.GitHub
+{
+    /// <summary>
+    /// Exposes methods to return GitHub API data
+    /// </summary>
+    public class GitHubService
+    {
+        public HttpClient Client { get; }
+
+        public GitHubService(HttpClient client)
+        {
+            client.BaseAddress = new Uri("https://api.github.com/");
+            client.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json"); // GitHub API versioning
+            client.DefaultRequestHeaders.Add("User-Agent", "HttpClientFactory-Sample"); // GitHub requires a user-agent
+
+            Client = client;
+        }
+
+        public async Task<GitHubIssue> GetLatestDocsIssue()
+        {
+            using (var stream = await Client.GetStreamAsync("/repos/aspnet/docs/issues?state=open&sort=created&direction=desc"))
+            using (var streamReader = new StreamReader(stream))
```

Comment:
I'd welcome thoughts on whether this is a) correct and b) too complicated for a sample?

I've done it this way as per the advice from @DamianEdwards and @davidfowl during NDC, they highlighted that for performance larger response content should be read as stream, rather than string. @DamianEdwards touched on this again during this week's community stand-up.

This seems to work and I hope is a reasonably valid approach but as per those sources, it's not necessarily easy to do right.

Personally I'd like the sample to try to be real world to some expect to avoid people copy/pasting demoware code, but I understand this may over-complicate the main topic being shown here.

cc// @scottaddie